objectivec/include/ort_session.h

```diff
@@ -46,14 +54,151 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * @param inputs Dictionary of input names to input ORT values.
  * @param outputNames Set of output names.
+ * @param runOptions Run configuration options.
  * @param[out] error Optional error information set if an error occurs.
  * @return A dictionary of output names to output ORT values with the outputs
  *         requested in `outputNames`, or nil if an error occurs.
  */
 - (nullable NSDictionary<NSString*, ORTValue*>*)runWithInputs:(NSDictionary<NSString*, ORTValue*>*)inputs
                                                   outputNames:(NSSet<NSString*>*)outputNames
+                                                   runOptions:(ORTRunOptions*)runOptions
                                                         error:(NSError**)error;
 
 @end
 
+/**
+ * Options for configuring a session.
+ */
+@interface ORTSessionOptions : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Creates session configuration options.
+ *
+ * @param[out] error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithError:(NSError**)error NS_SWIFT_NAME(init());
+
+/**
+ * Sets the number of threads used to parallelize the execution within nodes.
+ * A value of 0 means ORT will pick a default value.
+ *
+ * @param intraOpNumThreads The number of threads.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setIntraOpNumThreads:(int)intraOpNumThreads
+                       error:(NSError**)error;
```

Comment:
Curious why do you need to return BOOL as these functions already have the error param, I thought usually caller will check if the error is nil (seems to be the objc convention, but I could be wrong) to see if the function is executed correctly

Reply:
IIRC the bool return with an 'error' NSError param leads to the Swift bridge generation handling it nicely.

Reply:
methods returning BOOL will be imported as throwing methods in Swift
[https://github.com/apple/swift/blob/main/docs/CToSwiftNameTranslation.md#error-handling](https://github.com/apple/swift/blob/main/docs/CToSwiftNameTranslation.md#error-handling)